vpr/src/route/route_timing.cpp

```diff
@@ -1987,9 +1987,11 @@ bool is_iteration_complete(bool routing_is_feasible, const t_router_opts& router
     //in addition to routing being legal and the correct budgeting algorithm being set.
 
     if (routing_is_feasible) {
+        bool can_router_finish = rcv_finished || itry >= router_opts.max_router_iterations;
+
         if (router_opts.routing_budgets_algorithm != YOYO) {
             return true;
-        } else if (router_opts.routing_budgets_algorithm == YOYO && (timing_info->hold_worst_negative_slack() == 0 || rcv_finished) && itry != 1) {
+        } else if (router_opts.routing_budgets_algorithm == YOYO && (timing_info->hold_worst_negative_slack() == 0 || can_router_finish) && itry != 1) {
```

Comment:
Shouldn't the test for hold_worst_negative_slack be >= 0?  Seems a lot safer than a test for precisely 0.

Reply:
I would agree, but worst_negative_slack is equal to zero if worst hold slack is positive. Therefore they would equate to the same thing, but I could change it in case that changes in the future.

Reply:
OK, lets do >= for safety and readability though.